build/pkg.js

```diff
@@ -0,0 +1,116 @@
+const fsPromises = require("fs/promises");
+const fs = require("fs");
+
+const { parseManifest } = require("./cli.js");
+const { npm, npmInstall } = require("./npm.js");
+const { logDetail } = require("./log.js");
+
+function extractDependencies(manifest) {
+    const allDependencies = {
+        ...manifest.dependencies,
+        ...manifest.devDependencies,
+    };
+    const localDependencies = [];
+    const remoteDependencies = [];
+    for (let [name, url] of Object.entries(allDependencies)) {
+        const group = url.startsWith("file:")
+            ? localDependencies
+            : remoteDependencies;
+        group.push(name);
+    }
+    return { localDependencies, remoteDependencies };
+}
+
+class Pkg {
+    name;
+    path;
+    localDependencies;
+    remoteDependencies;
+    scripts;
+    _processed = false;
```

Comment:
I don't think these lines here do anything, you don't declare properties in JS, you simply set them in the constructor (or whenever you wanna set them).